crates/script/src/build.rs

```diff
@@ -133,17 +139,46 @@ impl LinkedBuildData {
     pub fn new(
         libraries: Libraries,
         predeploy_libraries: ScriptPredeployLibraries,
-        build_data: BuildData,
+        mut build_data: BuildData,
     ) -> Result<Self> {
         let sources = ContractSources::from_project_output(
             &build_data.output,
             &build_data.project_root,
             Some(&libraries),
         )?;
 
-        //TODO: zk contracts?
-        let known_contracts =
-            ContractsByArtifact::new(build_data.get_linker().get_linked_artifacts(&libraries)?);
+        let mut known_artifacts = build_data.get_linker().get_linked_artifacts(&libraries)?;
+        // Extend known_artifacts with zk artifacts if available
+        if let Some(zk_output) = build_data.zk_output.take() {
```

Comment:
I guess we need this because the abi is missing from DualCompiledContracts, right?

Reply:
Yes